### PR TITLE
Added JavaOSC 0.3 External Dependency and implemented the OSCForwarder class

### LINKED ISSUE #173: Add the Networking functionalities (Forwarders as well as Receivers)
**Actual Behaviour**
* The networking subpackage is incomplete and lacks several essential classes like OSCForwarder, OSCReceiver, SerialForwarder, SerialPortInterface, and SerialPortReader.

**Expected Behaviour**
* The networking package should be completely ported both the forwarders and receivers which would include finding and adding the correct external dependencies for the android version of the application and porting all these classes correctly for android.
* This would help to carry out our tests either on the hardware or on the test dataset recorded from a real device as @CloudyPadmal mentioned [here](https://github.com/fossasia/neurolab-android/issues/152). 

**Classes to Port and Checklist of the work** 
- [x] Finding and adding the correct external dependencies for the Android version.
- [x] Adding forwarders - OSCForwarder, SerialForwarder
- [x] Adding receivers - OSCReceiver 
- [x] Adding other utility classes - SerialPortInterface, SerialPortReader

**Would you like to work on the issue?**
I would like to work on this issue.

### PR DESCRIPTION
**Pull request by suresh-gandhi**
Fixes #173 partly (Targeting checklist points 1 and 2).

**Changes**: 
* Added the external dependency JavaOSC (version 0.3) for providing the Open Sound Control functionalities.
* Implemented the OSCForwarder class in the networking subpackage.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications are done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

**Comment by suresh-gandhi** [comment]
@iamareebjamal Could you please review and merge the PR?

**Comment by adityastic** [comment]
@suresh-gandhi it's weekend, he deserves a break xD.

File: app/build.gradle [hunk 1]
```diff
@@ -44,4 +44,5 @@ dependencies {
     androidTestImplementation "com.android.support.test:runner:$rootProject.testRunnerRulesVersion"
     androidTestImplementation "com.android.support.test.espresso:espresso-core:$rootProject.espressoVersion"
     implementation 'com.github.iamareebjamal:jsyn:16.8.0a'
+    implementation group: 'com.illposed.osc', name: 'javaosc-core', version: '0.3'
```
Comment (iamareebjamal) [inline app/build.gradle]:
```suggestion
    implementation group: 'com.illposed.osc:javaosc-core:0.3'
```

**Comment by suresh-gandhi** [comment]
@iamareebjamal did the changes. Please review and merge. Thanks.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**